images/router/haproxy/reload-haproxy

```diff
@@ -44,10 +44,41 @@ function haproxyHealthCheck() {
 }
 
 
-# How many times to retry removal of the iptables rules (if requested at all)
-# It will sleep for 1/2 a second between attempts, so the time is retries / 2 secs
-retries=20
+# Pull the ports we are listening on
+ports=$(grep -E '^\s*bind\s+:[[:digit:]]+\w' "$config_file" | cut -f2 -d: | paste -d, -s)
+
+# This ACK eating rule came from https://www.haproxy.com/blog/truly-seamless-reloads-with-haproxy-no-more-hacks/
+# The rule is:
+#        \! --u32 '0>>22&0x3C@12>>26&0x3C@0&0x0=0:0'
+#
+# The variables in the state machine (man iptables-extensions)
+#   A == The pointer into the header
+#   C == A uint32 we can load from A and manipulate
+#
+#   0        -- Load bytes 0-3 into C
+#   >> 22    -- Right shift C 22 bits to get to IP header length (in 32 bit words, so we need to multiply by 4... so we shift less)
+#   &  0x3C  -- Mask C with 0x3C (0011 1100) to drop the surrounding junk (masking out the low bits since we are multiplying as above)
+#   @  12    -- Set A = C + 12 (so skip the IP header length in C and 12 bytes more) to land at the TCP data offset, and load 4 bytes into C
+#   >> 26    -- Shift C as above (multiplied by 4 again) to get the TCP data offset
+#   &  0x3C  -- Mask C with 0x3C (0011 1100) to drop the surrounding junk (masking out the low bits since we are multiplying as above)
+#   @  0     -- Set A = C + 0 (so skip the TCP offset length held in C) to land at the TCP payload and load 4 bytes into C
+#               This step may cause it to fail if there are not 4 bytes of payload to load (since any access outside the packet range fails)
+#               We want the failure, because it means it is a short packet.  The next rules will cause the match to always succeed... which
+#               means we will ignore the packet.
+#   &  0x0   -- Mask C with 0x0 (i.e. set it to 0)
+#   =  0:0   -- And compare C to 0 (which will always succeed, so we ignore the packet)
+readonly ack_eater_comment='Eat empty ACKs while reloading haproxy'
+readonly ack_eater_rule=(-p tcp                                                   # Apply to TCP packets only
+                         -m multiport --dports ${ports}                           # On http and https ports (as determined above)
```

Comment:
We set it above to 80,443 (for instance).  It will always be number,number (and potentially ,number repeating if they have a customized router template).  So... it should not be expandable.  But should I wrap it in "" to be safe?